app/models/concerns/has_attachments.rb

```diff
@@ -1,17 +1,17 @@
 module HasAttachments
-    extend ActiveSupport::Concern
+  extend ActiveSupport::Concern
 
-    included do
-        validate :attachment_count,                                 :if => :published?
-    end
+  included do
+    validate :attachment_count, if: :published?, on: :create
```

Comment:
@malachaifrazier We would also need to trigger this validation on `:update` too if an admin decides to make a product a `draft`, then later on change it to a `published` state.